hudi-client/hudi-java-client/src/main/java/org/apache/hudi/client/HoodieJavaTableServiceClient.java

```diff
@@ -42,27 +48,31 @@ protected HoodieJavaTableServiceClient(HoodieEngineContext context,
   }
 
   @Override
-  protected HoodieWriteMetadata<List<WriteStatus>> compact(String compactionInstantTime, boolean shouldComplete) {
-    throw new HoodieNotSupportedException("Compact is not supported in HoodieJavaTableServiceClient");
+  protected void validateClusteringCommit(HoodieWriteMetadata<List<WriteStatus>> clusteringMetadata, String clusteringCommitTime, HoodieTable table) {
```

Comment:
we can probably keep this in BaseHoodieTableServiceClient only and leave it as protected. so that if any engine implements cluster, can use this method to validate.

Reply:
Wouldn't validation be the same across clients though?

Reply:
yes. you are right.